lib/outside/off/off_shops_manager.dart

```diff
@@ -54,6 +54,21 @@ class OffShopsManager {
     _offShopsOp.result.then((offShops) => offShops.maybeOk()?.dispose());
   }
 
+  Future<Result<OffShop, OffShopsManagerError>> findOffShopByName(
+      String name) async {
+    final shopsRes = await _offShopsOp.result;
+    if (shopsRes.isErr) {
+      return Err(shopsRes.unwrapErr());
+    }
+    final offShops = await shopsRes.unwrap().findAppropriateShopsFor([name]);
+    if (offShops.isEmpty) {
+      Log.w('offShopsManager.findOffShopByName: '
+          'No offshops found for name $name');
+      return Err(OffShopsManagerError.OTHER);
```

Comment:
Do you think maybe we should return `null` in cases when a shop is not found instead of errors?
Function's return type would have the nullable `OffShop?` instead of `OffShop` then.

Reply:
yes, i am not sure what the best solution was here. Null is ok for me.